### PR TITLE
hold `Options` instance in `DB` instance

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Hold `Options` instance in `DB` instance, so we can call function `get_statistics` of `Options` from `DB` instance.
@siddontang @BusyJay PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: src/rocksdb.rs [hunk 1]
```diff
@@ -294,61 +295,66 @@ impl DB {
...
+                .map(|cf| CString::new(cf.as_bytes()).unwrap())
+                .collect();
+
+            let cfnames: Vec<*const _> = c_cfs.iter()
+                .map(|cf| cf.as_ptr())
+                .collect();
+
+            // These handles will be populated by DB.
+            let cfhandles: Vec<_> = cfs_v.iter()
+                .map(|_| 0 as *mut DBCFHandle)
```
Comment (BusyJay) [inline src/rocksdb.rs]: use `ptr::null_mut()` instead.

**Comment by BusyJay** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**